scripts/skills/perks/perk_rf_supporter.nut

```diff
@@ -37,27 +38,30 @@ this.perk_rf_supporter <- this.inherit("scripts/skills/skill", {
 
 // MSU Functions
 	// Implement Action Point recovery after using a skill on adjacent allies
-	function onAnySkillExecuted( _skill, _targetTile, _targetEntity, _forFree )
+	// We measure the tile distance in onBeforeAnySkillExecuted because some skills e.g. Rotation
+	// may cause the actor to not be placed on map, causing tile distance to crash during onAnySkillExecuted.
+	// We then recover the AP in onAnySkillExecuted so the AP is recovered after spending AP on skill use.
+	function onBeforeAnySkillExecuted( _skill, _targetTile, _targetEntity, _forFree )
 	{
 		if (this.m.IsSpent || _targetEntity == null)
-		{
 			return;
-		}
 
 		local actor = this.getContainer().getActor();
-		if (actor.getFaction() == _targetEntity.getFaction() && actor.getID() != _targetEntity.getID())
+		if (actor.getFaction() != _targetEntity.getFaction() || actor.getID() == _targetEntity.getID() || !actor.isPlacedOnMap() || actor.getTile().getDistanceTo(_targetTile) > this.m.MinDistanceAPRecovery)
+			return;
+
+		this.m.RecoveredActionPoints = ::Math.min(actor.getActionPointsMax() - actor.getActionPoints(), this.m.ActionPointsRecovered);
```

Comment:
By this time the character has not yet spent any action points, so this will often times recover no action points.